### PR TITLE
test(rxPaginate): Exercises

### PR DESCRIPTION
**Pull request by Droogans**
Includes changes to the component template task, and many other grunt
build tasks in order to facilitate the use of page object
exercises. These are used to quickly generate known sets of (typically
simple, boring) tests to ensure that your encore components are
correctly wired up on the page.

### CODE REVIEW

**Comment by Droogans** [comment]
I'm now in a full-blown rebase frenzy between this, generating docs, and npm publishing.

@parlarjb please let me know when you're ready to get these all merged, since they're related. I'll need to do them in a certain order, or perhaps re-submit as one large PR once the individual ones have been cleared.

**Comment by jj4th** [comment]
@Droogans Build problem!!!

File: grunt-tasks/options/concat.js [hunk 1]
```diff
@@ -1,4 +1,11 @@
-var pageObjectHeader = '/*jshint node:true*/\nvar Page = require(\'astrolabe\').Page;\n';
+var removeFromIndex = ['/*jshint node:true*/\n',
+                       'var _ = require(\'lodash\');\n',
+                       'var moment = require(\'moment\');\n',
+                       'var Page = require(\'astrolabe\').Page;\n',
+                       'var exercise = require(\'./exercise\')\n'];
+
+var removeFromExercises = ['/*jshint node:true*/\n',
+                           'var _ = require(\'lodash\');\n'];
```
Comment (parlarjb) [inline grunt-tasks/options/concat.js]: Please document

Comment (Droogans) [inline grunt-tasks/options/concat.js]: Done. It's in [guides/testing.md](https://github.com/rackerlabs/encore-ui/pull/797/files#diff-f4e211adc2a9d601d0da67c5a548b199).

**Pull request closed without merging**